src/main/java/io/qameta/allure/maven/PropertyUtils.java

```diff
@@ -0,0 +1,62 @@
+package io.qameta.allure.maven;
+
+import org.apache.maven.shared.utils.StringUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Properties util class.
+ */
+class PropertyUtils {
+
+    /**
+     * Replaces the placeholders in properties.
+     * You can use properties like:
+     * name1=value1
+     * name2=value2 with ${name1}
+     * You can also use system and maven properties for the placeholder.
+     */
+    static void prepareProperties(Properties baseProperties) {
+        for (String name : baseProperties.stringPropertyNames()) {
+            baseProperties.setProperty(name, getPropertyValue(baseProperties.getProperty(name), baseProperties));
+        }
+    }
+
+    private static String getPropertyValue(String property, Properties baseProperties ) {
```

Comment:
I don't really understand what is going on in this method even it is copy-pasted from Maven sources